src/python/zquantum/qcbm/cost_function.py

```diff
@@ -52,7 +52,8 @@ def cost_function(
             (float): cost function value for given parameters
             zquantum.core.bitstring_distribution.BitstringDistribution: distribution obtained
         """
-        circuit = ansatz.get_executable_circuit(parameters)
+        # circuit = ansatz.get_executable_circuit(parameters)
+        circuit = ansatz._generate_circuit(parameters)
```

Comment:
Can you explain more why you chose to do this? If it's an efficiency improvement/avoiding some bottleneck, it would be better if we could instead figure out how to avoid the bottleneck in the first place as it's generally not good practice to call private methods from outside the class itself.

Reply:
I shared a word doc on quantum software channel on slack at the beginning of May with all the data and explanation for this :) Will send you the link

Reply:
I would suggest adding a comment with TODO here explaining why we're doing this.
Having TODO will help us remember to actually change this once we have proper solution in place.

Reply:
Added it myself ;)